rushti.py

```diff
@@ -330,12 +330,16 @@ def extract_tasks_from_file_type_opt(
                     tasks[task.id].append(task)
 
     # expand tasks
+    expanded_tasks = dict()
     if expand:
         for task_id in tasks:
             for task in tasks[task_id]:
                 for expanded_task in expand_task(tm1_services, task):
-                    tasks[task.id].append(expanded_task)
-                tasks[task.id].remove(task)
+                    if task.id not in expanded_tasks:
+                        expanded_tasks[task.id] = [expanded_task]
+                    else:
+                        expanded_tasks[task.id].append(expanded_task)
+        tasks = expanded_tasks
```

Comment:
Good catch!

Please add a test case here. This way, we can be 100% sure that the issue is resolved and if someone accidentally breaks it in the future, we will detect it
https://github.com/cubewise-code/rushti/blob/master/tests/tests.py